pkg/registry/core/nextwrap/ns_registry_client.go

```diff
@@ -40,10 +41,13 @@ func (r *nextNetworkServiceWrappedClient) Register(ctx context.Context, in *regi
 }
 
 func (r *nextNetworkServiceWrappedClient) Find(ctx context.Context, in *registry.NetworkServiceQuery, opts ...grpc.CallOption) (registry.NetworkServiceRegistry_FindClient, error) {
-	_, err := r.client.Find(ctx, in, opts...)
-	if err != nil {
+	client, err := r.client.Find(ctx, in, opts...)
+	if err != nil && err != io.EOF {
 		return nil, err
 	}
+	if client != nil {
```

Comment:
I think we need to have aggregator from two streams here.

Reply:
```go
type nsAggrClient struct {
	grpc.ClientStream
	ctx     context.Context
	streams []registry.NetworkServiceRegistry_FindClient
}

func (nsAggr *nsAggrClient) Recv() (*registry.NetworkService, error) {
	for {
		if nsAggr.streams == nil {
			return nil, io.EOF
		}
		res, err := nsAggr.streams[0].Recv()
		if err == io.EOF {
			nsAggr.streams = nsAggr.streams[1:]
			continue
		}
		return res, nil
	}
	return nil, nil
}

func (nsAggr *nsAggrClient) Context() context.Context {
	return nsAggr.ctx
}

func (r *nextNetworkServiceWrappedClient) Find(ctx context.Context, in *registry.NetworkServiceQuery, opts ...grpc.CallOption) (registry.NetworkServiceRegistry_FindClient, error) {
	clientClient, err := r.client.Find(ctx, in, opts...)
	if err != nil {
		return nil, err
	}
	var nextClient registry.NetworkServiceRegistry_FindClient
	nextClient, err = next.NetworkServiceRegistryClient(ctx).Find(ctx, in, opts...)
	if err != nil {
		return nil, err
	}
	return &nsAggrClient{
		ctx:     ctx,
		streams: []registry.NetworkServiceRegistry_FindClient{clientClient, nextClient},
	}, nil
}
```

Thoughts?

Reply:
@haiodo thanks, I think it is more correct direction. I'll improve ur suggestion to support monitoring cases.